### PR TITLE
Documentation: Add arity, correct arity, and link to callbacks and types

### PR DESCRIPTION
**Pull request by eksperimental**
Continuation of https://github.com/elixir-lang/elixir/pull/5301

### CODE REVIEW

File: lib/elixir/lib/code.ex [hunk 1]
```diff
@@ -247,9 +247,9 @@ defmodule Code do
   ## Options
 
     * `:file` - the filename to be used in stacktraces
-      and the file reported in the `__ENV__` variable
+      and the file reported in the `__ENV__/0` variable
```
Comment (michalmuskala) [inline lib/elixir/lib/code.ex]: Shouldn't it be "the `__ENV__/0` macro"?

File: lib/elixir/lib/code.ex [hunk 2]
```diff
@@ -247,9 +247,9 @@ defmodule Code do
...
 
-    * `:line` - the line reported in the `__ENV__` variable
+    * `:line` - the line reported in the `__ENV__/0` variable
```
Comment (michalmuskala) [inline lib/elixir/lib/code.ex]: Similar here

File: lib/elixir/lib/code.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/code.ex]: Yes!

Comment (eksperimental) [inline lib/elixir/lib/code.ex]: great catch @michalmuskala 
fixed!

File: lib/elixir/lib/code.ex (hunk 2 above)
Comment (eksperimental) [inline lib/elixir/lib/code.ex]: fixed. thank you

**Comment by whatyouhide** [comment]
Great job @eksperimental! 👍

**Comment by eksperimental** [comment]
thank you guys

**Pull request merged**